### PR TITLE
Feat/search filters refactor

### PR DESCRIPTION
**Pull request by matthieu-foucault**

### CODE REVIEW

File: app/components/FilterableTable/FilterableTablePagination.tsx [hunk 1]
```diff
@@ -87,19 +72,18 @@ const FilterableTablePaginationComponent: React.FunctionComponent<Props> = (
...
-              100
-            </Dropdown.Item>
+            {PAGE_SIZE_OPTIONS.map((option) => (
+              <Dropdown.Item
+                ref={option}
+                onClick={() => onPageSizeChange(option)}
+              >
+                {option}
+              </Dropdown.Item>
+            ))}
```
Comment (pbastia) [inline app/components/FilterableTable/FilterableTablePagination.tsx:86]: Since this is static we should compute it outside of the rendering loop

Comment (matthieu-foucault) [reply app/components/FilterableTable/FilterableTablePagination.tsx:86]: `onPageSizeChange` isn't static though

File: app/components/FilterableTable/FilterableTable.tsx [hunk 2]
```diff
@@ -1,41 +1,50 @@
...
-    () => safeJsonParse(router.query.relayVars as string),
+  const filterArgs = useMemo<FilterArgs>(
+    () => safeJsonParse(router.query.filterArgs as string),
+    [router]
+  );
+
+  const {offset, pageSize} = useMemo<PageArgs>(
+    () => safeJsonParse(router.query.pageArgs as string),
     [router]
   );
```
[the remark this answers is not in the document]
Comment (matthieu-foucault) [reply app/components/FilterableTable/FilterableTable.tsx:36]: We actually can rerender when the data changes, when adding products for instance. The impact either way is marginal

File: app/components/FilterableTable/FilterableTablePagination.tsx (hunk 1 above)
Comment (pbastia) [reply app/components/FilterableTable/FilterableTablePagination.tsx:86]: oh right!

**Review by pbastia — APPROVED** [review]

**Pull request merged**